src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -0,0 +1,389 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="HttpClientRequestMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client.Serialization
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.OData.Client;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using Microsoft.OData;
+
+    /// <summary>
+    /// HttpClient based implementation of DataServiceClientRequestMessage.
+    /// </summary>
+    public class HttpClientRequestMessage : DataServiceClientRequestMessage, IDisposable
+    {
+        /// <summary>
+        /// HttpClient distinguishes "Content" headers from "Request" headers, so we
+        /// need to know which category a header belongs to.
+        /// </summary>
+        private static readonly IEnumerable<string> contentHeaderNames = new[] {
+            "Allow",
+            "Content-Disposition",
+            "Content-Encoding",
+            "Content-Language",
+            "Content-Length",
+            "Content-Location",
+            "Content-MD5",
+            "Content-Range",
+            "Content-Type",
+            "Expires",
+            "Last-Modified",
+        };
+
+        /// <summary>Request Url.</summary>
+        private readonly Uri requestUrl;

```

Comment:
Consider using `_fieldName` for field names and drop the use of `this.fieldName` in the rest of the class

Reply:
resolved